service/src/main/java/tds/config/services/impl/ConfigServiceImpl.java

```diff
@@ -24,16 +26,35 @@ public ConfigServiceImpl(ConfigRepository configRepository) {
 
     @Override
     public Optional<ClientTestProperty> getClientTestProperty(final String clientName, final String assessmentId) {
-        ClientTestProperty clientTestProperty = configRepository.getClientTestProperty(clientName, assessmentId);
-
-        return Optional.ofNullable(clientTestProperty);
+        return configRepository.getClientTestProperty(clientName, assessmentId);
     }
 
     @Override
     public Optional<ClientSystemFlag> getClientSystemFlag(String clientName, String auditObject) {
         List<ClientSystemFlag> clientSystemFlags = configRepository.getClientSystemFlags(clientName);
 
-        return clientSystemFlags.stream().filter(f -> f.getAuditObject().equals(auditObject))
+        return clientSystemFlags.stream()
+                .filter(f -> f.getAuditObject().equals(auditObject))
                 .findFirst();
     }
+
+    @Override
+    public Optional<Integer> getTaCheckInTimeLimit(final String clientName) {
+        Optional<TimeLimits> timeLimits =  configRepository.getTimeLimits(clientName);
+
+        if (timeLimits.isPresent()) {
+            return Optional.of(
+                    timeLimits.map(TimeLimits::getTaCheckinTimeMinutes)
+                    .orElse(0));
```

Comment:
Looking at the database, it's possible that the `tacheckin` column is `null`.  The legacy code sets the `tacheckin` time to `0` in the event it's null (or not returned at all, if I'm not mistaken).  This is a sanity check more than anything; the legacy logic seems to say the `tacheckin` should always be something, so I'm trying to carry that forward.  A comment about that in the code would probably be helpful.